### PR TITLE
Bug1625151 5.6

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
http://jenkins.percona.com/job/percona-server-5.6-param/1570/
http://jenkins.percona.com/job/percona-server-5.6-param/1571/

### CODE REVIEW

File: mysql-test/suite/innodb/t/bug84366.test [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                 INDEX i1(key1), INDEX i2(key2)) ENGINE = InnoDB;
+
+--disable_query_log
+INSERT INTO t1 VALUES (1, 1), (2, 2);
+
+let $1=9;
+SET @d=2;
+while ($1)
+{
+  eval INSERT INTO t1 SELECT key1 + @d, key2 + @d FROM t1;
```
Comment (percona-ysorokin) [inline mysql-test/suite/innodb/t/bug84366.test]: unnecessary 'eval'

File: mysql-test/suite/innodb/t/bug84366.test [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  eval set @d=@d*2;
```
Comment (percona-ysorokin) [inline mysql-test/suite/innodb/t/bug84366.test]: unnecessary 'eval'

File: mysql-test/suite/innodb/t/bug84366.test [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+connection default;
+SET DEBUG_SYNC = "now WAIT_FOR estimate_ready";
+
+SET @@GLOBAL.innodb_purge_run_now = TRUE;
+
+# wait_innodb_all_purged.inc is not used directly as we wait for the trx age
+# to become one, not zero, due to one open transaction
+--let $status_var= innodb_purge_trx_id_age
+--let $status_var_value= 1
+
```
Comment (percona-ysorokin) [inline mysql-test/suite/innodb/t/bug84366.test:53]: either missing 'wait_for_status_var.inc' or unnecessary '$status_var' and '$status_var_value' assignments.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Comment by laurynas-biveinis** [comment]
http://jenkins.percona.com/job/percona-server-5.6-param/1575/

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**